installers/metrics/ansible/roles/pgo-metrics/tasks/grafana.yml

```diff
@@ -84,15 +82,12 @@
 
     - name: Create Grafana Dashboard ConfigMap
       shell: |
-        {{ kubectl_or_oc }} create configmap {{ grafana_dashboards_configmap }} \
+        {{ kubectl_or_oc }} create configmap {{ grafana_dashboards_configmap }} --dry-run --output=yaml \
           --from-file={{ grafana_output_dir }}/crunchy_grafana_dashboards.yml \
-          --from-file={{ grafana_dashboards | join(' --from-file=') }} \
-          -n {{ metrics_namespace }} --dry-run -o yaml | \
-            {{ kubectl_or_oc }} label configmap --local vendor=crunchydata -f - \
-              -n {{ metrics_namespace }} --dry-run -o yaml | \
-                {{ kubectl_or_oc }} create -f - -n {{ metrics_namespace }}
-      args:
-        chdir: "{{ pgmonitor_grafana_dir }}"
+          --from-file={{ pgmonitor_grafana_dir }}/containers/ \
+        | sed -e 's,${DS_PROMETHEUS},PROMETHEUS,' \
```

Comment:
Just curious if the `sed` step is really needed?

When I install without this step, Grafana seems to handle the replacement, which [per the docs](https://grafana.com/docs/grafana/latest/reference/export_import/#import-and-sharing-with-grafana-2x-or-30) seems to be because the following is present:

```json
"__inputs": [
    {
      "name": "DS_PROMETHEUS",
      "label": "PROMETHEUS",
      "description": "",
      "type": "datasource",
      "pluginId": "prometheus",
      "pluginName": "Prometheus"
    }
```

End result is the same either way, just seems interesting if their Operator is doing what Grafana itself already appears to do itself :thinking:

Reply:
`DS_PROMETHEUS` is included in `__inputs` when a dashboard is exported "for sharing externally". When the dashboard is _imported_, this field causes Grafana to prompt the user to select a Prometheus data source. Unfortunately, however, Grafana provides [no way to make that selection when _provisioning_ dashboards](//github.com/grafana/grafana/issues/10786).